core/src/main/java/org/opensearch/sql/datasource/model/DataSourceType.java

```diff
@@ -56,4 +57,17 @@ public static DataSourceType fromString(String name) {
       throw new IllegalArgumentException("No DataSourceType with name " + name + " found");
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    DataSourceType that = (DataSourceType) o;
+    return Objects.equals(name, that.name);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(name);
+  }
```

Comment:
Can we use `@EqualsAndHashCode` annotation instead?

If we need our own implementation, we should add unit tests.

Reply:
changed to use `@EqualsAndHashCode` annotation